sw/device/tests/power_virus_systemtest.c

```diff
@@ -451,7 +480,114 @@ static void configure_spi_host() {
   CHECK_DIF_OK(dif_spi_host_output_set_enabled(&spi_host_0, /*enabled=*/true));
 }
 
+static void crypto_data_load_task(void *task_parameters) {
+  LOG_INFO("Loading crypto block FIFOs with data ...");
+  // Load data into AES block.
+  dif_aes_data_t aes_plain_text;
+  memcpy(aes_plain_text.data, kAesModesPlainText, sizeof(aes_plain_text.data));
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusInputReady, true,
+                                kTestTimeoutMicros);
+  CHECK_DIF_OK(dif_aes_load_data(&aes, aes_plain_text));
+
+  // Load data into HMAC block.
+  hmac_testutils_push_message(&hmac, kHmacRefData, sizeof(kHmacRefData));
+  hmac_testutils_check_message_length(&hmac, sizeof(kHmacRefData) * 8);
+
+  // Load data into KMAC block.
+  CHECK_DIF_OK(dif_kmac_absorb(&kmac, &kmac_operation_state, kKmacMessage,
+                               kKmacMessageLength, NULL));
+
+  OTTF_TASK_DELETE_SELF_OR_DIE;
+}
+
+static void comms_data_load_task(void *task_parameters) {
+  LOG_INFO("Loading communication block FIFOs with data ...");
+  size_t bytes_written;
+  CHECK(ARRAYSIZE(kUartMessage) == 32);
+  dif_uart_t uart_handles[] = {uart_1, uart_2, uart_3};
+
+  // Load data into UART FIFO.
+  for (size_t i = 0; i < ARRAYSIZE(uart_handles); ++i) {
+    bytes_written = 0;
+    CHECK_DIF_OK(dif_uart_bytes_send(&uart_handles[i], kUartMessage,
+                                     ARRAYSIZE(kUartMessage), &bytes_written));
+    CHECK(bytes_written == ARRAYSIZE(kUartMessage));
+  }
+
+  OTTF_TASK_DELETE_SELF_OR_DIE;
+}
+
+static void max_power_task(void *task_parameters) {
+  LOG_INFO("Starting the max power task ...");
+  // ***************************************************************************
+  // Trigger all chip operations.
+  //
+  // Note: We trigger the activations of each operation manually, rather than
+  // use the DIFs, so that we can maximize the time overlap between all
+  // operations.
+  // ***************************************************************************
+
+  // Prepare KMAC for squeeze command. Note, below code is derived from the KMAC
+  // DIF `dif_kmac_sqeeze()`.
+  CHECK(!kmac_operation_state.squeezing);
```

Comment:
Not sure if it's important for this test to use KMAC specifically, but if you just want to actuate the Keccak core it might be easier to use e.g. SHAKE so you don't have to use a key or do the `append_d` step.

Reply:
The goal here was to activate as much of the data path as possible, i.e. key, customization string etc, hence KMAC mode. I'm guessing the Keccak core itself will dominate power usage, especially since we are mostly just doing some extra padding, IIUC no? But would you say that operating in KMAC mode represents the most intensive use all hw components of the KMAC block?